src/api.cr

```diff
@@ -22,6 +23,19 @@ module OpenTelemetry
           ],
         )
       end
+
+      def resource_from_env
+        return nil if ENV["HONEYCOMB_DATASET"]?.nil?
+
+        Proto::Resource::V1::Resource.new(
+          attributes: [
+            Proto::Common::V1::KeyValue.new(
+              key: "service.name",
+              value: Proto::Common::V1::AnyValue.new(ENV["HONEYCOMB_DATASET"])
```

Comment:
I was even thinking how this ENV variable could be plucked out from here and only use the config (when initializing projects) instead, in case that sounds better with something like this:

```crystal
OpenTelemetry.configure do |c|
    c.exporter = OpenTelemetry::BatchExporter.new(
      OpenTelemetry::HTTPExporter.new(
        endpoint: URI.parse("https://api.honeycomb.io"),
        headers: HTTP::Headers{
          # Get your Honeycomb API key from https://ui.honeycomb.io/account
          "x-honeycomb-team"    => ENV["HONEYCOMB_API_KEY"]
        },
      )
    )
    c.service_name = ENV["HONEYCOMB_DATASET"]
  end
```

Not sure what sounds/looks better 🤔